staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -0,0 +1,357 @@
+type ValidatingAdmissionPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// List of ValidatingAdmissionPolicy.
+	Items []ValidatingAdmissionPolicy `json:"items,omitempty" protobuf:"bytes,2,rep,name=items"`
+}
+
+// ValidatingAdmissionPolicySpec is the specification of the desired behavior of the AdmissionPolicy.
+type ValidatingAdmissionPolicySpec struct {
+	//ParamSource specifies the kind of resources used to parameterize this policy which has to be a CRD.
+	// If absent, there are no parameters for this policy and the param CEL variable will not be provided to validation expressions.
+	// If ParamSource refers to a non-existent kind, this policy definition is mis-configured and the FailurePolicy is applied.
+	// +optional
+	ParamSource *ParamSource `json:"paramSource,omitempty" protobuf:"bytes,1,rep,name=paramSource"`
+
+	// MatchConstraints specifies what resources this policy is designed to validate.
+	// The AdmissionPolicy cares about a validation if it matches _any_ Constriant.
+	// However, in order to prevent clusters from being put into an unstable state that cannot be recovered from via the API
+	// ValidatingAdmissionPolicy cannot match ValidatingAdmissionPolicy/ValidatingAdmissionPolicyBinding/param resources.
+	// ValidatingWebhookConfiguration cannot match MutatingWebhookConfiguration or ValidatingAdmissionPolicy/PolicyBinding/param resources.
+	// Required.
+	MatchConstraints *MatchResources `json:"matchConstraints,omitempty" protobuf:"bytes,2,rep,name=matchConstraints"`
+
+	// Validations contain CEL expressions which is used to apply the validation.
+	// A minimum of one validation is required for a policy definition.
+	// Required.
+	Validations []Validation `json:"validations" protobuf:"bytes,3,rep,name=validations"`
+
+	// FailurePolicy defines how to handle failures for the admission policy.
+	// Failures can occur from invalid policy definitions or bindings, as well as validation errors from CEL expressions.
+	// A policy is invalid if spec.paramSource refers to a non-existent Kind.
+	// A binding is invalid if spec.paramName refers to a non-existent custom resource.
+	// +optional
+	FailurePolicy *FailurePolicyType `json:"failurePolicy,omitempty" protobuf:"bytes,4,opt,name=failurePolicy,casttype=FailurePolicyType"`
+}
+
+// ParamSource is a tuple of Group Kind and Version.
+type ParamSource struct {
+	// APIVersion is the API group version the resources belong to.
+	// In format of "group/version".
+	// Required.
+	APIVersion string `json:"apiVersion,omitempty" protobuf:"bytes,1,rep,name=apiVersion"`
+
+	// APIKind is the API kind the resources belong to.
+	// Required.
+	APIKind string `json:"apiKind,omitempty" protobuf:"bytes,2,rep,name=apiKind"`
```

Comment:
This should simply be called "kind" to match metadata and the other parts of the API that identify via apiVersion/kind fields.